src/r-bridge/lang-4.x/ast/parser/xml/internal/loops/for.ts

```diff
@@ -88,7 +88,7 @@ function normalizeForHead(data: ParserData, forCondition: XmlBasedJson): { varia
 	const vector = normalizeBasedOnType(data, [others[inPosition + 1]])
 	guard(vector.length === 1 && vector[0].type !== RType.Delimiter, () => `for loop vector should have been parsed to a single element but was ${JSON.stringify(vector)}`)
 	const parsedComments: (RNode | RDelimiter)[] = comments.map(c => normalizeComment(data, c.content))
-	parsedComments.push(...normalizeBasedOnType(data, [others[0]])) // e.g., push '('
+	parsedComments.push(...normalizeBasedOnType(data, others.filter(token => token.name === RawRType.ForIn || token.name === RawRType.ParenLeft || token.name === RawRType.ParenRight))) // e.g., push '('
```

Comment:
You can not use `normalizeBasedOnType` for `ForIn` as there is no handler for it. You have to parse it as a new delimiter.
In `tryNormalizeSingleNode` (`single-element.ts`) you must add a handler for `RawRType.ForIn` (use `normalizeDelimiter` as a basis).